### DISCUSSION TITLE
update  current User password redirect  wrong

### DISCUSSION THREAD

**Discussion by oz-hjx210** [filamentphp/filament#6203] [Help]
When I modify the current user's password in FILAMENT ADMIN at admin/profile, the system automatically logs me out, and refreshing the page redirects me to the front-end login page (/login) instead of /admin/login. Where should I configure this redirect? thanks

**Comment by leandrocfe** [comment]
This isn't the default behavior. Are you using Filament v3? What is the Filament version on the dashboard page?

**Comment by SirFaenor** [comment]
This is related to this Laravel middleware `Illuminate/Session/Middleware/AuthenticateSession.php` that checks that the hashed password in session is the same of the logged user and trigger a logout if the check fails:
```php
if ($request->session()->get('password_hash_'.$this->auth->getDefaultDriver()) !== $request->user()->getAuthPassword()) 
    $this->logout($request);
}
```
So you can change the default redirect for auth middleware or avoid logging out the user by updating the value in session, just after user has been updated (check Filament docs):
```php
session()->put([
   'password_hash_web' => $userThatHasBeenJustUpdated->password
]);
```
where the 'web' part of the string is the name of the guard you are using.

Edit: typo, code block.
